tensorflow_io/core/kernels/http_kernels.cc

```diff
@@ -13,16 +13,527 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
-// TODO: Replace curl_http_request.h to not depends on TensorFlow's
-// C++ implementation
-#include "tensorflow/core/platform/cloud/curl_http_request.h"
+#include <curl/curl.h>
+
+#include <iostream>
+#include <string>
+#include <unordered_map>
+
+#include "absl/strings/ascii.h"
+#include "absl/strings/str_cat.h"
+#include "absl/synchronization/mutex.h"
 #include "tensorflow_io/core/kernels/file_system_plugins.h"
 
 namespace tensorflow {
 namespace io {
 namespace http {
 namespace {
 
+// Set to 1 to enable verbose debug output from curl.
+constexpr uint64_t kVerboseOutput = 0;
+
+static absl::Mutex mu;
+static bool uninitialized(false);
+void CurlInitialize() {
+  absl::MutexLock l(&mu);
+  if (!uninitialized) {
+    curl_global_init(CURL_GLOBAL_ALL);
+    uninitialized = true;
```

Comment:
@yongtang I think using `initialized` instead of `uninitialized` is more relatable.

```cpp
static absl::Mutex mu;
static bool initialized(false);
void CurlInitialize() {
  absl::MutexLock l(&mu);
  if (!initialized) {
    curl_global_init(CURL_GLOBAL_ALL);
    initialized = true;
  }
}
```

Reply:
@kvignesh1420 Updated. Thanks.